### PR TITLE
Simplify create process for into AbstractMultiBranchCreateRequest

### PR DESCRIPTION
**Pull request by i386**
# Description

This change makes it very easy for people to add create requests for multibranch pipelines with non-git SCMSources 

See https://github.com/jenkinsci/blueocean-plugin/pull/1037 for an implementation.

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/AbstractPipelineCreateRequest.java [hunk 1]
```diff
@@ -41,6 +48,6 @@
         if(!acl.hasCreatePermission(a, p, descriptor)){
             throw new ServiceException.ForbiddenException("Missing permission: " +Item.CREATE.group.title+"/"+Item.CREATE.name + Item.CREATE + "/" + descriptor.getDisplayName());
         }
-        return parent.createProject(descriptor, name, true);
+        return Jenkins.getInstance().createProject(descriptor, name, true);
```
Comment (vivek) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/AbstractPipelineCreateRequest.java]: `parent.createProject()` is fine, at the abstract level, better for client to decide where to create this project.

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/AbstractMultiBranchCreateRequest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Creates {@link MultiBranchProject}s with a single {@link SCMSource}
+ */
+public abstract class AbstractMultiBranchCreateRequest extends AbstractPipelineCreateRequest {
```
Comment (vivek) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/AbstractMultiBranchCreateRequest.java]: 👍

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/AbstractPipelineCreateRequest.java (hunk 1 above)
Comment (i386) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/AbstractPipelineCreateRequest.java]: @vivek can always replace that call with a 
```
Folder getItemRoot() { ... }
```
... in the abstract class and ask the client to set it as a string on posting this request?

Comment (vivek) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/AbstractPipelineCreateRequest.java]: That works too.

File: blueocean-pipeline-api/src/main/java/io/jenkins/blueocean/pipeline/creation/AbstractMultiBranchCreateRequest.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
+package io.jenkins.blueocean.pipeline.creation;
```
Comment (vivek) [inline blueocean-pipeline-api/src/main/java/io/jenkins/blueocean/pipeline/creation/AbstractMultiBranchCreateRequest.java]: Guess all these could be inside `api` package, like `io.jenkins.blueocean.pipeline.api`?

Comment (i386) [inline blueocean-pipeline-api/src/main/java/io/jenkins/blueocean/pipeline/creation/AbstractMultiBranchCreateRequest.java]: @vivek sounds reasonable. Ive made the change. I also moved the credentials back into `io/jenkins/blueocean/rest/impl/pipeline/credential` package as I think they their fully qualified names get baked into xstream

Comment (vivek) [inline blueocean-pipeline-api/src/main/java/io/jenkins/blueocean/pipeline/creation/AbstractMultiBranchCreateRequest.java]: Hmm,  I am going to check. We are using credentials apis for everything, why are our classes getting in there? I will check sometime tonight or tomorrow morning.

**Comment by i386** [comment]
@vivek I think our Credential classes need to be moved out to another module

**Comment by i386** [comment]
ATH https://ci.blueocean.io/blue/organizations/jenkins/ATH-Jenkinsfile/detail/master/2376/pipeline

**Comment by vivek** [comment]
@i386 LGTM 🐝 We can move credentials in to another module in separate PR.

**Pull request merged**